### ISSUE TITLE
[Bug]: Warning: Cannot load system font: AdobeSongStd-Light, installing it could help to improve PDF rendering.

### ISSUE THREAD

**Issue by plainnany** [mozilla/pdf.js#18362]
### Attach (recommended) or Link to PDF file

[problem.pdf](https://github.com/user-attachments/files/16061281/problem.pdf)

### Web browser and its version

chrome 126.0.6478.127

### Operating system and its version

MacOS Monterey 12.6

### PDF.js version

4.4.170

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1. https://mozilla.github.io/pdf.js/web/viewer.html upload the file
2. some text disappeared, and the console output “Warning: Cannot load system font: AdobeSongStd-Light, installing it could help to improve PDF rendering.”
<img width="1255" alt="demo" src="https://github.com/mozilla/pdf.js/assets/27320655/a24f2a09-8453-4ff2-8322-1d1656004d3b">

### What is the expected behavior?

render normal

### What went wrong?

some text disappeared、

### Additional context

how to add new custom font？https://github.com/adobe-type-tools/cmap-resources/tree/master/ in adobe cmap-resources,I can't find AdobeSongStd-Light

**Comment by THausherr** [comment]
Yeah it's missing in `Root/Pages/Kids/[0]/Resources/XObject/Xi16/Resources/Font/AdobeSongStd-Light/DescendantFonts/[0]/FontDescriptor`. You'll have to buy it from Adobe, or ask the creator of the PDF ("hangzhou esign Group") to embed it to avoid troubles like this.

**Comment by timvandermeij** [comment]
Based on the above, I don't think there is much we can do here other than perhaps using a fallback font? Fonts should always be embedded in PDF files to avoid such issues.

**Comment by calixteman** [comment]
There's nothing we can do here.
If someone can provide a good free equivalent (in term of metrics), please provide us the font name.
